nav2_util/include/nav2_util/simple_action_server.hpp

```diff
@@ -220,38 +261,22 @@ class SimpleActionServer
     return current_handle_->is_canceling();
   }
 
-  void terminate_goals(
```

Comment:
Wouldn't it be easier to change this to `terminate_goal` and just end the current goal. By ending the current goal the pending goal can be processed. The SimpleActionServer only processes one goal at a time and there are 2 termination conditions: user asked for termination (and killing all is reasonable) and action failed to process and in that case you shouldn't penalize the pending actions.

Reply:
Will make the change.